addons/hr_holidays/tests/test_global_leaves.py

```diff
@@ -125,3 +126,45 @@ def test_global_leave_timezone(self):
         # Note:
         # The user in Europe/Brussels timezone see 4:30 and not 2:30 because he is in UTC +02:00.
         # The user in Asia/Calcutta timezone (determined via the browser) see 8:00 because he is in UTC +05:30
+
+    def test_global_leave_multi_companies(self):
+        other_company = self.env['res.company'].create({'name': 'Other company'})
+        calendar_other_company = self.env['resource.calendar'].create({
+            'name': 'Classic 40h/week Other company',
+            'tz': 'UTC',
+            'company_id': other_company.id,
+            'hours_per_day': 8.0,
+            'attendance_ids': [
+                (0, 0, {'name': 'Monday Morning', 'dayofweek': '0', 'hour_from': 8, 'hour_to': 12, 'day_period': 'morning'}),
+                (0, 0, {'name': 'Monday Afternoon', 'dayofweek': '0', 'hour_from': 13, 'hour_to': 17, 'day_period': 'afternoon'}),
+                (0, 0, {'name': 'Tuesday Morning', 'dayofweek': '1', 'hour_from': 8, 'hour_to': 12, 'day_period': 'morning'}),
+                (0, 0, {'name': 'Tuesday Afternoon', 'dayofweek': '1', 'hour_from': 13, 'hour_to': 17, 'day_period': 'afternoon'}),
+                (0, 0, {'name': 'Wednesday Morning', 'dayofweek': '2', 'hour_from': 8, 'hour_to': 12, 'day_period': 'morning'}),
+                (0, 0, {'name': 'Wednesday Afternoon', 'dayofweek': '2', 'hour_from': 13, 'hour_to': 17, 'day_period': 'afternoon'}),
+                (0, 0, {'name': 'Thursday Morning', 'dayofweek': '3', 'hour_from': 8, 'hour_to': 12, 'day_period': 'morning'}),
+                (0, 0, {'name': 'Thursday Afternoon', 'dayofweek': '3', 'hour_from': 13, 'hour_to': 17, 'day_period': 'afternoon'}),
+                (0, 0, {'name': 'Friday Morning', 'dayofweek': '4', 'hour_from': 8, 'hour_to': 12, 'day_period': 'morning'}),
+                (0, 0, {'name': 'Friday Afternoon', 'dayofweek': '4', 'hour_from': 13, 'hour_to': 17, 'day_period': 'afternoon'})
+            ]
+        })
+        other_company.resource_calendar_id = calendar_other_company
+        self.env['resource.calendar.leaves'].with_company(other_company).create({
+            'name': 'Global Leave Other company',
+            'date_from': "2022-03-15 00:00:00",
+            'date_to': "2022-03-15 23:59:59",
+        })
+        list_days = self.employee_emp.with_context(allowed_company_ids=[self.env.company.id, other_company.id], company_id=self.env.company.id)._get_unusual_days(date(2022, 3, 1), date(2022, 3, 30))
```

Comment:
Same for other lines :innocent: 
```suggestion
        list_days = self.employee_emp\
            .with_context(allowed_company_ids=[self.env.company.id, other_company.id], company_id=self.env.company.id)\
            ._get_unusual_days(date(2022, 3, 1), date(2022, 3, 30))
```